pkg/ocm/helpers.go

```diff
@@ -125,30 +125,88 @@ func ClusterDomainPrefixValidator(domainPrefix interface{}) error {
 	if str, ok := domainPrefix.(string); ok {
 		str := strings.Trim(str, " \t")
 		if str != "" && !IsValidClusterDomainPrefix(str) {
-			return fmt.Errorf("Cluster domain prefix must consist of no more than %d lowercase "+
+			return fmt.Errorf("cluster domain prefix must consist of no more than %d lowercase "+
 				"alphanumeric characters or '-', start with a letter, and end with an "+
-				"alphanumeric character.", MaxClusterDomainPrefixLength)
+				"alphanumeric character", MaxClusterDomainPrefixLength)
 		}
 		return nil
 	}
 	return fmt.Errorf("can only validate strings, got '%v'", domainPrefix)
 }
 
-func ValidateHTTPProxy(val interface{}) error {
-	if httpProxy, ok := val.(string); ok {
-		if httpProxy == "" {
-			return nil
-		}
-		url, err := url.ParseRequestURI(httpProxy)
-		if err != nil {
-			return fmt.Errorf("Invalid http-proxy value '%s'", httpProxy)
+// validateProxyURL validates common proxy URL requirements
+func validateProxyURL(proxyURL string, proxyType string, expectedScheme string) error {
+	if strings.Contains(proxyURL, " ") {
+		return fmt.Errorf("invalid %s value: URL cannot contain spaces", proxyType)
+	}
+
+	if credValidation := helper.ValidateURLCredentials(proxyURL); credValidation.HasInvalid {
+		return fmt.Errorf("invalid %s value: %s contains invalid character '%c'",
+			proxyType, credValidation.Field, credValidation.InvalidChar)
+	}
+
+	parsedURL, err := url.Parse(proxyURL)
+	if err != nil {
+		return fmt.Errorf("invalid %s value: %v", proxyType, err)
+	}
+
+	if parsedURL.Scheme == "" {
+		return fmt.Errorf("invalid %s value: URL scheme is missing (e.g., %s://)", proxyType, expectedScheme)
+	}
+	if parsedURL.Scheme != expectedScheme {
+		return errors.Errorf("%s", fmt.Sprintf("expected %s to have an %s:// scheme", proxyType, expectedScheme))
+	}
+
+	if parsedURL.Host == "" {
+		return fmt.Errorf("invalid %s value: host is missing", proxyType)
+	}
+
+	// Proxy URLs shouldn't have a path
+	if parsedURL.Path != "" && parsedURL.Path != "/" {
+		return fmt.Errorf("invalid %s value: proxy URL should not contain a path", proxyType)
+	}
+
+	// Validate hostname/IP and port
+	host, port, err := net.SplitHostPort(parsedURL.Host)
+	if err != nil {
+		// No port specified, just validate the host
+		host = parsedURL.Host
+	} else {
+		// Validate port range (url.Parse already validated it's numeric)
+		portNum, _ := strconv.Atoi(port)
+		if portNum < 1 || portNum > 65535 {
+			return fmt.Errorf("invalid %s value: port must be between 1 and 65535", proxyType)
```

Comment:
I don't think we need this validation specifically- the user shouldn't be inputting a giant or negative port number if they have a proxy URL they know they want to use. Having it result in a 404 eventually would be fine UX. WDYT?